phpBB/includes/functions.php

```diff
@@ -70,10 +70,18 @@ function set_var(&$result, $var, $type, $multibyte = false)
 *
 * @return string
 */
-function gen_rand_string($num_chars = 8)
+function gen_rand_string($num_chars = 8, $characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890')
 {
-	// [a, z] + [0, 9] = 36
-	return substr(strtoupper(base_convert(bin2hex(random_bytes($num_chars)), 16, 36)), 0, $num_chars);
```

Comment:
Why not just change this to `substr(strtoupper(base_convert(bin2hex(random_bytes($num_chars + 1)), 16, 36)), 0, $num_chars);`? Should work just fine and you'll not be using something that should be actually random.

Reply:
My version is better, the old one fails with big quantities of characters too

Reply:
The expected length fails with 65+ characters with your modification, also base_convert function starts to fail with 127+ characters